### ISSUE TITLE
Failing serialization of geometrycollection of multi to GeoJSON

### ISSUE THREAD

**Issue by ThomasG77** [TWKB/twkb.js#11]
I was trying to use twkt to serialize a geometrycollection of multipolygons. At first, I was looking at the serialization on client side. After more investigation, it seems this is not managed correctly (or as I expect it).

The proof? See this additional tests on my fork https://github.com/webgeodatavore/twkb.js/commit/8d0e638bb4646aa0b568942de4a3d77e1feb2498#diff-20139373d7d13bbc72485ffbafed376dR108
I've included 2 tests. The first one fails with a "TypeError: geoms.map is not a function"

IMHO, the spec is not really clear about allowing to nest a "multi" within a geometrycollection. So, I don't know if the issue is my interpretation or in the code.

If you consider my issue as valid, I will make a PR with my commit to introduce the 2 additional tests including the failing one and will investigate to fix (already some clues but do not want to fix before it's confirmed)

**Comment by nicklasaven** [comment]
As I read the spec a geometry collection contains any geometry, which should include also multi or collections. I do not recall exactly those discussions, but I am pretty sure it is handled in PostGIS implementation at least. 

I will take a look.

**Comment by ThomasG77** [comment]
> I am pretty sure it is handled in PostGIS implementation

I can confirm this point because I've documented [the PostGIS query](https://github.com/webgeodatavore/twkb.js/commit/8d0e638bb4646aa0b568942de4a3d77e1feb2498#diff-20139373d7d13bbc72485ffbafed376dR108) to produce the value in the tests.

**Comment by bjornharrtell** [comment]
Agreed this looks like a bug in twkb.js implementation.

**Comment by bjornharrtell** [comment]
Harder to fix than expected. Problem is deeper, the TWKB spec allows nested "features" (geom(s) with id) which is not allowed in GeoJSON.

**Comment by bjornharrtell** [comment]
Have put considerable thought into the matter and there is simply no way to properly map TWKB to GeoJSON unless some very strict rules are emposed on TWKB i.e it cannot be nested. Not sure how to proceed.

**Comment by ThomasG77** [comment]
As they can't match 1 to 1 properly, the only solution is to abandon the idea and mention the limitation at least.
The other question is concerning what the spec should say about this use case.
Should it say that nested GeometryCollection should not be used as it can lead to failure when serializing to other formats (without mentioning GeoJSON as it's a spec)

**Comment by nicklasaven** [comment]
I agree that it should be mentioned in twkbjs that since the output is geojson it doesn't suppott nested multi or collections. 

But I disagree that this should affect any recommendations for twkb. That is why we have different formats. Twkb was actually designed to be more flexible when it comes to nesting and relations than other formats. That is why it doesn't include attributes more than id. For 1:1 mapping to geojson I guess geobuf is a better fit.

**Comment by ThomasG77** [comment]
@nicklasaven It's fine for me. So, what the statement on the twkbjs side should be?
> Beware that if a geometry collection is nested or contains a "multi" geometry, it will not be possible to convert it to GeoJSON?

**Comment by bjornharrtell** [comment]
Some things can be done on twkb.js yet - proper error handling/message for the unsupported case and possibly also support for the the case where we have a flat collection of multigeoms without ids (should work but doesn't with current implementation).
